relay-server/src/actors/project_buffer.rs

```diff
@@ -519,14 +546,22 @@ impl Drop for BufferService {
         let count: usize = self.buffer.values().map(|v| v.len()).sum();
         // We have envelopes in memory, try to buffer them to the disk.
         if count > 0 {
-            if let Err(err) = self.try_spool() {
-                relay_log::error!("failed to spool {} on shutdown: {}", count, LogError(&err));
-            }
-
-            // The buffer must be empty by now, report the error otherwise.
-            let count: usize = self.buffer.values().map(|v| v.len()).sum();
-            if count > 0 {
-                relay_log::error!("dropped {} envelopes", count);
+            let buffer = std::mem::take(&mut self.buffer);
+            if let Some(BufferSpoolConfig { db, .. }) = &self.spool_config {
+                let db = db.clone();
+                tokio::spawn(async move {
+                    let Some(mut connection) = db.try_acquire() else {
+                        relay_log::error!("failed to get connection while dropping the project buffer");
+                        return;
+                    };
+                    if let Err(err) = Self::do_spool(&mut connection, buffer).await {
```

Comment:
We drop this service mostly when the runtime stops, which means we should no longer start async tasks. Also, our expectation would be that when we drop the service, we close the DB and it's connections, which now isn't true because the DB lives on in a spawned task. 

This kind of procedure should probably be implemented in response to the `shutdown` signal.

For the MVP, it's enough if we drop the buffer and log outcomes + errors.